pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -153,12 +154,33 @@ func MakeClusterSpec(nodeCount int, opts ...Option) ClusterSpec {
 
 // ClustersCompatible returns true if the clusters are compatible, i.e. the test
 // asking for s2 can reuse s1.
-func ClustersCompatible(s1, s2 ClusterSpec) bool {
-	s1.Lifetime = 0
-	s2.Lifetime = 0
+func ClustersCompatible(s1, s2 ClusterSpec, cloud string) bool {
+	// only consider the specification of the cloud that we are running in
+	clearClusterSpecFields(&s1, cloud)
```

Comment:
Pardon my ignorance... why did we have to resort to a rather hacky way of implementing equality _modulo_? We're already using `cmp.Equal` in places, e.g., [1]. It's a much cleaner way, imo; specifically, it's side-effect free.

[1] https://github.com/cockroachdb/cockroach/blob/3bd657275574961162a702c9407f45f885fdd0e0/pkg/sql/sem/tree/backup.go#L390